apf/apf.cc

```diff
@@ -163,13 +161,15 @@ void destroyField(Field* f)
 
 void setScalar(Field* f, MeshEntity* e, int node, double value)
 {
+  PCU_DEBUG_ASSERT(f->getValueType() == SCALAR);
```

Comment:
If we are now allowing c++11 in CORE can this be done with a static assert?

Reply:
Core has been C++11 for a while IIRC. I've been using `auto` and range-based for's for a while no one has complained so far.

This assert has to be runtime though.